### PR TITLE
Sui: Transfer tokens with payload, dynamic maps, VAA tests, checks, refactoring, Worm CLI start

### PR DESCRIPTION
**Pull request by optke3**
## Updates
- Implement `transfer_tokens_with_payload` and make `transfer_tokens` an entry function
- Make `treasury_cap` and `coin_store` dynamic maps take `CoinTypeWrapper` as Key-type rather than `OriginInfo`
- Remove `chainID` and `governance_contract` from token bridge state
- Allow wormhole and token_bridge `init_and_share_state` take in vector of initial guardians instead of just one
- Added more checks, assertions, comments, custom error codes, and tests throughout the codebase
- Write Sui-style unit tests for token bridge `VAA`
- Reconsider and evaluate remaining bottlenecks / features we’re waiting on from Mysten Labs
- Get started on Typescript Sui Worm CLI

## New Assumptions
- a Sui-native token needs to be attested / registered with token bridge before it can be transferred to token bridge

### CODE REVIEW

File: sui/token_bridge/sources/bridge_state.move [hunk 1]
```diff
@@ -32,10 +30,40 @@ module token_bridge::bridge_state {
...
 
-   /// TODO - The origin chain and address of a token.  In case of native tokens
-   ///        what do we set the token_address to? For Aptos it was the hash of the
-   ///        deployer + module_name + struct_name, for Sui might have to do differently
+   /// WrappedAssetInfo is a Sui object (has key ability) that functions as a Value
+   /// in a dynamic object mapping that maps Names => Values. It stores metadata about
+   /// a particular CoinType, or equivalently a CoinTypeWrapper.
+   struct WrappedAssetInfo has key, store {
+      id: UID,
+      token_chain: U16,
```
Comment (kcsongor) [inline sui/token_bridge/sources/bridge_state.move:44]: why not put an `OriginInfo` in here? (same for `NativeAssetInfo`)

File: sui/wormhole/sources/state.move [hunk 2]
```diff
@@ -87,14 +87,23 @@ module wormhole::state {
...
         set_governance_contract(&mut state, governance_contract);
-        let initial_guardian = vector[structs::create_guardian(initial_guardian)];
-        store_guardian_set(&mut state, u32::from_u64(0), structs::create_guardian_set(u32::from_u64(0), initial_guardian));
+
+        let guardians = vector::empty<Guardian>();
+        let i = 0;
+        while (i < vector::length<vector<u8>>(&initial_guardians)){
+            vector::push_back<Guardian>(&mut guardians, structs::create_guardian(*vector::borrow<vector<u8>>(&initial_guardians, i)));
+            i = i + 1;
+        };
```
Comment (kcsongor) [inline sui/wormhole/sources/state.move:102]: I think it'd be good to keep code as consistent with the aptos version as we can
https://github.com/wormhole-foundation/wormhole/blob/5c0ed392d464d02de1d4bcb2d5ac4e4cedcb7a2c/aptos/wormhole/sources/wormhole.move#L70-L73

File: sui/token_bridge/sources/bridge_state.move (hunk 1 above)
Comment (optke3) [reply sui/token_bridge/sources/bridge_state.move:44]: I can change this later.

File: sui/wormhole/sources/state.move (hunk 2 above)
Comment (optke3) [reply sui/wormhole/sources/state.move:101]: changed, thanks

**Pull request merged**